doit/dependency.py

```diff
@@ -536,13 +555,27 @@ def get_status(self, task, tasks_dict):
 
         @param task: (Task)
         @param tasks_dict: (dict: Task) passed to objects used on uptodate
-        @return: (str) one of up-to-date, run
+        @param get_log: (bool) if True, adds all reasons to the return
+                               object why this file will be rebuild.
+        @return: (DependencyStatus) a status object with possible status
+                                    values up-to-date, run or error
 
         task.dep_changed (list-strings): file-dependencies that are not
         up-to-date if task not up-to-date because of a target, returned value
         will contain all file-dependencies reagrdless they are up-to-date
         or not.
         """
+        result = DependencyStatus()
+        if get_log:
+            result.uptodate_false = []
```

Comment:
Since you created a class `DependencyStatus` this initialization should be done in `DependencyStatus.__init__()`